ql_demos/cpp/libssh2_eating_error_codes/README.md

```diff
@@ -0,0 +1,9 @@
+# Eating error codes in libssh2
+
+Use this [Snapshot](https://downloads.lgtm.com/snapshots/cpp/libssh2/libssh2_libssh2_C_C++_38bf7ce.zip) for the demo.
+
+This demo shows how to develop, step-by-step, the query from the [blog post](https://blog.semmle.com/libssh2-integer-overflow/) about libssh2 CVE-2019-13115. This query did not find the bug that caused the CVE. It is instead about doing variant analysis on a bug which we noticed on the development branch of libssh2. We sent the query results to the libssh2 development team and they were able to fix all the variants before the next version of libssh2 was released.
```

Comment:
```suggestion
This demo shows how to develop, step-by-step, the query from the [blog post](https://blog.semmle.com/libssh2-integer-overflow/) about libssh2 CVE-2019-13115. This query did not find the bug that caused the CVE. It is instead about doing variant analysis on a bug that we noticed on the development branch of libssh2. We sent the query results to the libssh2 development team and they were able to fix all the variants before the next version of libssh2 was released.
```

Reply:
I am happy to accept this change, but please could you explain why "that" is better than "which" here? ("which vs that" is something that I am a bit fuzzy on.)

Reply:
I follow the rationale outlined here: https://www.writersdigest.com/online-editor/which-vs-that

As I understand it, if you are using 'which' you should be able to drop the clause without changing the meaning of the sentence. That doesn't seem to be the case here, as we need the clause to identify the bug specifically as the one on the development branch. Without the clause the sentence doesn't make much sense.

[As a rough rule, 'which' would normally follow a comma too.]

It's one of those things that probably doesn't matter too much, but I've got into the habit of spotting and correcting. I'm sure you could get away without making the change.

Hope that helps?

Reply:
Thanks!